server/protocols/jmap/src/main/java/org/apache/james/jmap/crypto/DEREncodingConverter.java

```diff
@@ -0,0 +1,24 @@
+package org.apache.james.jmap.crypto;
+
+import org.bouncycastle.openssl.PEMReader;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Optional;
+
+public class DEREncodingConverter {
+
+    Optional<RSAPublicKey> fromPEM(Optional<String> pemKey) {
+
+        return pemKey
+                .map(k -> new PEMReader(new StringReader(k)))
+                .map(r -> {
+                    try {
+                        return (RSAPublicKey) r.readObject();
+                    } catch (IOException e) {
+                        return null;
```

Comment:
null ? so you return Option or null depending on the error type ?
I think you should not catch that IOException at all, but I'm not really sure